src/lib.rs

```diff
@@ -1046,8 +1033,7 @@ impl Node {
 	/// Will also remove the peer from the peer store, i.e., after this has been called we won't
 	/// try to reconnect on restart.
 	pub fn disconnect(&self, counterparty_node_id: PublicKey) -> Result<(), Error> {
-		let rt_lock = self.runtime.read().unwrap();
-		if rt_lock.is_none() {
+		if !*self.is_running.read().unwrap() {
```

Comment:
Out of scope for this PR, but curious: we talked about the type-state pattern for `Runtime` and that it may not be ideal. Could the pattern work for `Node` though? So start returning a type that exposes the method, and that type being consumed by stop.

Reply:
I see. I have no experience with the pattern. It looks pretty powerful in combination with Rust's type system. Don't know if the pipe through can be done with a wrapper around the state types. Either way, this was just a side remark.